src/lib/mina_net2/libp2p_stream.ml

```diff
@@ -142,10 +176,13 @@ let create_from_existing ~logger ~helper ~stream_id ~protocol ~peer
   in
   let send_outgoing_messages_task =
     Pipe.iter outgoing_r ~f:(fun msg ->
+        let parts = split_string msg ~every:max_message_size in
         match%map
-          Libp2p_helper.do_rpc helper
-            (module Libp2p_ipc.Rpcs.SendStream)
-            (Libp2p_ipc.Rpcs.SendStream.create_request ~stream_id ~data:msg)
+          Deferred.Or_error.List.iter parts ~f:(fun data ->
```

Comment:
Nit: We're creating and then immediately iterating over a `list` here. Wouldn't it be more efficient to iterate over the sting using the steps directly?